operator/pkg/securedcluster/scanner/auto_sense_test.go

```diff
@@ -53,3 +64,35 @@ func TestAutoSenseIsEnabledWithCentralInADifferentNamespace(t *testing.T) {
 	require.NoError(t, err)
 	require.True(t, enabled, "Expected Scanner to be enabled if Central is deployed in a different namespace")
 }
+
+func TestAutoSenseIsDisabledIfClusterVersionNotFound(t *testing.T) {
+	client := testutils.NewFakeClientBuilder(t, &osconfigv1.ClusterVersion{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: testutils.TestNamespace,
```

Comment:
Can you try this test if the namespace is missing as the `ClusterVersion` is a global resource?
I am not sure if the underlying kubernetes fake client supports these features as well.

Reply:
I dropped `Namespace` from the search key. Thus missing namespace is a default case. Although I added test case when `ClusterVersion` has namespace. It's pretty unlikely and it's definitely something fishy so test asserts that such `ClusterVersion` will not be found and hence OpenShift will not be detected.

Reply:
You don't need to test `ClusterVersion` with namespace and without, this is can't happen in a production cluster. I meant a manual test that the fake k8s client respects these API features.
Sorry if I did not communicated it clearly enough.